src/Microsoft.FeatureManagement/FeatureFilters/Cron/CronExpression.cs

```diff
@@ -0,0 +1,121 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+//
+using System;
+
+namespace Microsoft.FeatureManagement.FeatureFilters.Cron
+{
+    /// <summary>
+    /// The Cron expression with Minute, Hour, Day of month, Month, Day of week fields.
+    /// </summary>
+    internal class CronExpression
+    {
+        private static readonly int NumberOfFields = 5;
+
+        private readonly CronField _minute;
+        private readonly CronField _hour;
+        private readonly CronField _dayOfMonth;
+        private readonly CronField _month;
+        private readonly CronField _dayOfWeek;
+
+        private CronExpression(CronField minute, CronField hour, CronField dayOfMonth, CronField month, CronField dayOfWeek)
+        {
+            _minute = minute;
+            _hour = hour;
+            _dayOfMonth = dayOfMonth;
+            _month = month;
+            _dayOfWeek = dayOfWeek;
+        }
+
+        /// <summary>
+        /// Check whether the given expression can be parsed by a CronExpression.
+        /// If the expression is invalid, an ArgumentException will be thrown.
+        /// </summary>
+        /// <param name="expression">The expression to parse.</param>
+        /// <returns>A parsed CronExpression.</returns>
+        public static CronExpression Parse(string expression)
+        {
+            if (expression == null)
+            {
+                throw new ArgumentNullException(nameof(expression));
+            }
+
+            string InvalidCronExpressionErrorMessage = $"The provided Cron expression: '{expression}' is invalid.";
+
+            var fields = new string[NumberOfFields];
+
+            int i = 0, pos = 0;
+
+            while (i < expression.Length)
+            {
+                if (char.IsWhiteSpace(expression[i]))
+                {
+                    i++;
+
+                    continue;
+                }
+
+                if (pos >= fields.Length)
+                {
+                    throw new ArgumentException(InvalidCronExpressionErrorMessage, nameof(expression));
+                }
+
+                int start = i; // Start of a field
+
+                while (i < expression.Length && !char.IsWhiteSpace(expression[i]))
+                {
+                    i++;
+                }
+
+                fields[pos] = expression.Substring(start, i - start);
```

Comment:
Even better if we can use `ReadOnlySpan<char>` to avoid allocating new strings.

Reply:
CronField.TryParse(CronFieldKind kind, string content, out CronField result) will take a string as parameter. This method will also call string.Split() and string.Contains(). Should I change them and use ReadOnlySpan<char> instead?